airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -42,6 +43,36 @@ class SlackStream(HttpStream, ABC):
     url_base = "https://slack.com/api/"
     primary_key = "id"
     page_size = 100
+    # We need different retry behaviour for different response types, implement
+    # own retry counter.
+    retry_attempts = 0
+    max_retry_attempts = 5
+    # Since there is no limit on retry count for 429 response we should limit
+    # it by time to avoid endless loop.
+    retry_timeout = pendulum.duration(minutes=5)
+    request_started_at = None
+
+    @property
+    def max_retries(self) -> Union[int, None]:
```

Comment:
Resetting some side variables based on accessing a property is very hacky/non-obvious and could result in incorrect behavior. 

Is there any reason not to just increase retries a lot? e.g: `max_retries = 20` and call it a day? is there a need for all the extra logic? 

We could implement the thing you are trying to do here correctly by first implementing https://github.com/airbytehq/airbyte/issues/2762, but the current approach is using the retry methods incorrectly

Reply:
@sherifnada Done, just increased max retry attempts